src/game/WorldHandlers/LootHandler.cpp

```diff
@@ -255,15 +255,26 @@ void WorldSession::HandleLootMoneyOpcode(WorldPacket& /*recv_data*/)
 
             break;
         }
-        case HIGHGUID_CORPSE:                               // remove insignia ONLY in BG
+        /* HACK: Due to the spaghetti code below, */
+        /* we have a special case here for the looting of player corpses in battlegrounds. */
+        case HIGHGUID_CORPSE:
         {
             Corpse* bones = _player->GetMap()->GetCorpse(guid);
 
             if (bones && bones->IsWithinDistInMap(_player, INTERACTION_DISTANCE))
             {
                 pLoot = &bones->loot;
-            }
+                pLoot->NotifyMoneyRemoved();
+                player->ModifyMoney(pLoot->gold);
 
+                // Used by Eluna
+                #ifdef ENABLE_ELUNA
+                    sEluna->OnLootMoney(player, pLoot->gold);
+                #endif /* ENABLE_ELUNA */
+
+                pLoot->gold = 0;
+                return;
+            }
```

Comment:
Lets try to avoid hacks when fixing bugs. The extra code is unneeded.

Here's a suggestion for the fix to prevent crashing. Might need to adjust the `shareMoney = false` bool.
```suggestion
        case HIGHGUID_CORPSE:                               // remove insignia ONLY in BG
        {
            Corpse* bones = ObjectAccessor::GetCorpse(*player, guid);

            if (bones && bones->IsWithinDistInMap(player, INTERACTION_DISTANCE))
            {
                loot = &bones->loot;
                shareMoney = false;
            }

            break;
        }
```

Reply:
I fail to see how this fixes anything. Isn't it just the code as it was prior to my commit?